### PR TITLE
[cmake] More work on OSX

### PR DESCRIPTION
**Pull request by fetzerch**
@Memphiz: fyi, loading the dylibs works now.

### CODE REVIEW

File: project/cmake/scripts/darwin/macros.cmake [hunk 1]
```diff
@@ -48,34 +48,51 @@ function(core_link_library lib wraplib)
...
+                  COMMAND sed "s/-L[ ]*//g"
+                  COMMAND sed "s/-l[^ ]*//g"
+                  ERROR_QUIET
+                  OUTPUT_VARIABLE env_lib_path
+                  OUTPUT_STRIP_TRAILING_WHITESPACE)
+  if(CMAKE_CROSSCOMPILING)
+    set(host_lib_path /usr/lib /usr/local/lib)
   endif()
+
+  foreach(path ${cc_lib_path} ${env_lib_path} ${host_lib_path})
```
Comment (Memphiz) [inline project/cmake/scripts/darwin/macros.cmake]: what if the lib is in our depends and in host libdir? (curl is such a candidate - we need to ensure to take the one from depends and not from the system) - not even sure if we would ever dyload a lib from the host path directly tbh.

Comment (fetzerch) [inline project/cmake/scripts/darwin/macros.cmake]: I was not sure about that one either. I kept it as it was done that way in the configure script. If I'm not mistaken, CMAKE_CROSSCOMPILING is anyways not set in the Toolchain file so we'd never add the host paths. I'll remove that lines.

**Comment by FernetMenta** [comment]
sorry, I missed this and forced pushed already. could you rebase please

File: project/cmake/scripts/darwin/macros.cmake (hunk 1 above)
Comment (akva2) [inline project/cmake/scripts/darwin/macros.cmake]: the whole cmake build was, at least from my side, designed around the idea of only a single toolchain in the buildsystem. anything for host was meant to be done in a separate pre-step (currently handled by tools/depends). that way you can completely jail the build, which minimizes the chance of any host stuff slipping through the cracks.

**Comment by fetzerch** [comment]
rebased and removed the host related part from find_soname

**Pull request merged**